### PR TITLE
Feature/developer roles

### PR DESCRIPTION
**Pull request by andrewkrug**
- add cross account roles
- add dynamodb and kinesis stream 
- setup Dynamo Stream for loopback and identity drivers.

### CODE REVIEW

**Comment by andrewkrug** [comment]
Updated cloudformation stack to align with suggestion in last PR .  Now matches library convention for attribute names.

File: cloudformation/PolicyTemplates/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+type lambda, kinesis, or dynamo tagged in this manner will allow explicit access.
+
+Recommended that you use this with aws-mfa python module for compatibility with Apex framework.
+
+To use aws-mfa start by:
+
+    pip install -r requirements.txt
+    Edit your ~/.aws/credentials file and create a profile that looks like this with static access keys that give you the ability to assume_role.
+
+[defult-long-term]
```
Comment (gdestuynder) [inline cloudformation/PolicyTemplates/README.md]: typo ;-)

File: cloudformation/PolicyTemplates/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+aws_access_key_id = YOUR_LONGTERM_KEY_ID
+aws_secret_access_key = YOUR_LONGTERM_ACCESS_KEY
+
+I set up a bash alias for convenience after this that looks like:
+
+alias cisaws="aws-mfa --device {YOUR_MFA_ARN} --assume-role arn:aws:iam::656532927350:role/CISDeveloper --role-session-name "cis-developer-session""
+
+> Note you still need to put in your MFA DEVICE ARN
+
+My creds generated in this manner are good for 60-minutes. I could set up additional aliases for additional profiles if necessary.
```
Comment (gdestuynder) [inline cloudformation/PolicyTemplates/README.md]: just for clarity i would change the I/My to "the credentials" "Setup up..." etc.

File: cloudformation/PolicyTemplates/01-cis-developer-policy.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    Properties:
+      Description: Allows developers working on change integration admin level access to resources with specific tags.
+      Path: /
+      PolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          -
+            Effect: Allow
+            Action: lambda:*
+            Resource: "arn:aws:lambda:*:*:function:cis_functions*"
```
Comment (johngian) [inline cloudformation/PolicyTemplates/01-cis-developer-policy.yml]: Although i think that this works just fine why ``cis_functions*`` and not ``*cis_functions*`` ?

File: cloudformation/PolicyTemplates/03-cross-account-policy.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+AWSTemplateFormatVersion: "2010-09-09"
+Description: "Allows users in parsys or infosec prod to call assumeRole for CISDeveloper in infosec-dev."
```
Comment (johngian) [inline cloudformation/PolicyTemplates/03-cross-account-policy.yml]: How are you allowing parsys users to assume this rule? Are we manually configuring ``CISDeveloper`` group?

File: cloudformation/PolicyTemplates/04-apex-permissions.yml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+              - "lambda:GetAlias"
+              - "lambda:ListAliases"
+              - "lambda:ListVersionsByFunction"
+            Resource: "arn:aws:lambda:*:*:function:cis_functions*"
+          -
+            Effect: Allow
+            Action:
+              - "logs:FilterLogEvents"
+              - "cloudwatch:GetMetricStatistics"
+            Resource: "*"
```
Comment (johngian) [inline cloudformation/PolicyTemplates/04-apex-permissions.yml]: Why do we allow ``apex`` access to all the resources?

**Comment by johngian** [comment]
@andrewkrug I approved it with some comments to avoid blocking it.

File: cloudformation/PolicyTemplates/04-apex-permissions.yml (hunk 5 above)
Comment (andrewkrug) [inline cloudformation/PolicyTemplates/04-apex-permissions.yml]: @johngian lambda functions need access to do those two specific permissions in order to search for metrics and retrieve cloudwatch events from any function in the project.  This is likely hardenable in the production deploy once we know what the naming convention is for groups that are created.  Allowing filterLogEvents and getmetricstats is effectively just read only on all lambda logs.

File: cloudformation/PolicyTemplates/03-cross-account-policy.yml (hunk 4 above)
Comment (andrewkrug) [inline cloudformation/PolicyTemplates/03-cross-account-policy.yml]: There is one permission that you have to place on Parsys accounts if non admin that allows them to assume the role.  Everything else is already setup with trust to PartInfra.

File: cloudformation/PolicyTemplates/01-cis-developer-policy.yml (hunk 3 above)
Comment (andrewkrug) [inline cloudformation/PolicyTemplates/01-cis-developer-policy.yml]: The lambda function names always start with the project name through declared convention.  We would have to override the behavior in apex's config to change this.  So it seemed fine to me.

**Pull request merged**